functions/_fzf_preview_status.fish

```diff
@@ -0,0 +1,24 @@
+function _fzf_preview_status
+    set -l path (string split ' ' $argv)[-1]
+    if string match -r '^\?\?' $argv >/dev/null
```

Comment:
having a bit of trouble following the logic of each if/else if/else. could you outline the "algorithm" at a high level and explain why you think it's comprehensive?

Reply:
1. use `_fzf_preview_file` to preview untracked files or dirs
2. diff with /dev/null and new files
3. then other files are indexed, then seperate unstaged and staged files as you mentioned.

I think that this basically covers common situations